geomstats/geometry/matrices.py

```diff
@@ -67,40 +93,49 @@ def transpose(mat):
         -------
         transpose : array-like, shape=[n_samples, dim, dim]
         """
-        is_vec = (gs.ndim(gs.array(mat)) == 3)
-        axes = (0, 2, 1) if is_vec else (1, 0)
+        is_vectorized = (gs.ndim(gs.array(mat)) == 3)
+        axes = (0, 2, 1) if is_vectorized else (1, 0)
         return gs.transpose(mat, axes)
 
-    @staticmethod
-    def vector_from_matrix(matrix):
-        """Conversion function from (_, m, n) to (_, mn)."""
-        matrix = gs.to_ndarray(matrix, to_ndim=3)
-        n_mats, m, n = matrix.shape
-        return gs.reshape(matrix, (n_mats, m*n))
+    @classmethod
+    def is_symmetric(cls, mat, atol=TOLERANCE):
+        """
+        Check if a matrix is symmetric.
+        
+        Parameters
+        ----------
+        mat : array-like, shape=[n_samples, n, n]
+        atol : float, absolute tolerance. defaults to TOLERANCE
 
-    @staticmethod
-    def is_symmetric(matrix, tolerance=TOLERANCE):
-        """Check if a matrix is symmetric."""
-        matrix = gs.to_ndarray(matrix, to_ndim=3)
-        n_mats, m, n = matrix.shape
-        assert m == n
-        matrix_transpose = gs.transpose(matrix, axes=(0, 2, 1))
+        Returns
+        -------
+        is_sym : array-like boolean, shape=[n_samples]
+        """
+        return cls.equal(mat, cls.transpose(mat), atol)
 
-        mask = gs.isclose(matrix, matrix_transpose, atol=tolerance)
-        mask = gs.all(mask, axis=(1, 2))
+    @classmethod
+    def make_symmetric(cls, mat):
+        """
+        Make a matrix symmetric, by averaging with its transpose.
 
-        mask = gs.to_ndarray(mask, to_ndim=1)
-        mask = gs.to_ndarray(mask, to_ndim=2, axis=1)
-        return mask
+        Parameters
+        ----------
+        mat : array-like, shape=[n_samples, n, n]
+        
+        Returns
+        -------
+        sym : array-like, shape=[n_samples, n, n]
+        """
+        return 1/2 * (mat + cls.transpose(mat))
 
     @staticmethod
-    def make_symmetric(matrix):
-        """Make a matrix fully symmetric to avoid numerical issues."""
+    def vector_from_matrix(matrix):
+        """
+        Conversion function from (_, m, n) to (_, mn).
+        """
         matrix = gs.to_ndarray(matrix, to_ndim=3)
         n_mats, m, n = matrix.shape
```

Comment:
I know it's not your code, but maybe `dim_1`, `dim_2` instead of `m, n` will be more consistent with the naming of this file + avoid single letter variable

Reply:
Ok I can change these, but as per @stefan's remarks I could as well delete all these vector to matrix methods

Reply:
I do agree it's not the right place for them, and if ever needed we could start putting those in separate mappings files

Reply:
yes, `vectorization.py` 's goal is to factorize these. #379